python/knext/knext/operator/spg_record.py

```diff
@@ -0,0 +1,251 @@
+        self, property_name: PropertyName, default_value: str = None
+    ) -> str:
+        """Gets a property of this SPGRecord by name.  # noqa: E501
+
+
+        :param property_name: The property name.  # noqa: E501
+        :param default_value: If property value is None, the default_value will be return.  # noqa: E501
+        :return: A property value.  # noqa: E501
+        :rtype: str
+        """
+        return self.properties.get(property_name, default_value)
+
+    def upsert_property(self, property_name: PropertyName, value: str):
+        """Upsert a property of this SPGRecord.  # noqa: E501
+
+
+        :param property_name: The updated property name.  # noqa: E501
+        :param value: The updated property value.  # noqa: E501
+        :type: str
+        """
+        self.properties[property_name] = value
+        return self
+
+    def upsert_properties(self, properties: Dict[PropertyName, str]):
+        """Upsert properties of this SPGRecord.  # noqa: E501
+
+
+        :param properties: The updated properties.  # noqa: E501
+        :type: dict
+        """
+        self.properties.update(properties)
+        return self
+
+    def remove_property(self, property_name: PropertyName):
+        """Removes a property of this SPGRecord.  # noqa: E501
+
+
+        :param property_name: The property name.  # noqa: E501
+        :type: str
+        """
+        self.properties.pop(property_name)
+        return self
+
+    def remove_properties(self, property_names: List[PropertyName]):
+        """Removes properties by given names.  # noqa: E501
+
+
+        :param property_names: A list of property names.  # noqa: E501
+        :type: list
+        """
+        for property_name in property_names:
+            self.properties.pop(property_name)
+        return self
+
+    def get_relation(
+        self,
+        relation_name: RelationName,
+        object_type_name: SPGTypeName,
+        default_value: str = None,
+    ) -> str:
+        """Gets a relation of this SPGRecord by name.  # noqa: E501
+
+
+        :param relation_name: The relation name.  # noqa: E501
+        :param object_type_name: The object SPG type name.  # noqa: E501
+        :param default_value: If property value is None, the default_value will be return.  # noqa: E501
+        :return: A relation value.  # noqa: E501
+        :rtype: str
+        """
+        return self.relations.get(relation_name + "#" + object_type_name, default_value)
+
+    def upsert_relation(
+        self, relation_name: RelationName, object_type_name: SPGTypeName, value: str
+    ):
+        """Upsert a relation of this SPGRecord.  # noqa: E501
+
+
+        :param relation_name: The updated relation name.  # noqa: E501
+        :param object_type_name: The object SPG type name.  # noqa: E501
+        :param value: The updated relation value.  # noqa: E501
+        :type: str
+        """
+        self.relations[relation_name + "#" + object_type_name] = value
```

Comment:
add verification logic for the existence of the triple `<_spg_type_name, relation_name, object_type_name>` in SPGSchema.